### ISSUE TITLE
Taking over

### ISSUE THREAD

**Issue by dalefrancis88** [tycho01/pointfree#1]
Hey,

This looks to be no longer maintained but i like the name and would like to do something very similiar to this. Would you consider letting me take this over?

**Comment by KiaraGrouwstra** [comment]
@dalefrancis88 hi! I've added you as a collaborator to give you push access. :)
what I felt was missing here was proper typing, but I fear TypeScript at least is still far away from being able to do that...
so I didn't really know where to take this from here anymore myself. pretty curious on your take!

**Comment by dalefrancis88** [comment]
Well i may not be able to help out much there. For me i find the surge of typescript ludicrous, one of the great powers of JS is it's lack of enforced types in my opinion :) FP concepts make that even more true. 

Personally i was thinking of taking this in a whole different direction of making deliberate pointfree methods for each method on all the types. When it comes to external libraries my thought was people could offer contrib for the smaller ones and the larger ones i was thinking scoped packages

**Comment by dalefrancis88** [comment]
I think the focus was more to take over the name on npm and the repo, as opposed to contributing, but i can understand the idea, given that this is OS :)

**Comment by CrossEye** [comment]
@dalefrancis88: If you're interested in an entirely different direction, it would probably be better to start a new repo, and probably use a different NPM name.

**Comment by KiaraGrouwstra** [comment]
@dalefrancis88
perhaps they can share repo so that both approaches are exposed?
otherwise, if yours is more popular then yielding the npm name might make sense.

manually handling each function separately huh... what issue would this address? like, optional arguments still clash with currying right? I'm trying to imagine what your interface would then look like.

**Comment by dalefrancis88** [comment]
So kinda like how you want types, i'd want the inputs to be validated. I'm also against the runtime creation/scan. Without any evidence to back me up, i feel that the approach could lead to unforseen side effects.

Sure, i've started to set something up, when it's up and running i'll shoot it through

**Comment by KiaraGrouwstra** [comment]
you mean run-time checks in the spirit of Sanctuary? sounds pretty interesting!

> Without any evidence to back me up, i feel that the approach could lead to unforeseen side effects.

hm, while this was more of a PoC and I haven't actually used it much, I think the approach here was actually simple enough to be predictable.
so I tried doing two things here:
- currying. while partly delegated to Ramda, the challenge here was conciliating currying with JS's unpredictable function arity due to optional arguments and variadic functions. I tried to use JS's built-in function param count check, but that would only count the mandatory args, ignoring optionals/variadics. as a result, currying here is only partial: you'd need to pass the last mandatory arg (where applicable) and all optionals/variadics in one go, or it'd think it already had enough.
- flip prototype subjects / first arg for functions to the last position to enable point-free style. this was also a challenge due to the optionals/variadics mentioned above. it felt hard to enable both calls passing and not passing the optionals and have it know what was the final (data) argument. I addressed this by forcing you to put the data arg in separate parentheses, i.e. `fn(my, args, optional)(data)`. the alternative was pre-specified arities e.g. `fn4(my, args, optional, data)` or `fn3(my, args, data)`, in which case it would know how many optionals to expect.

I thiiink any automation involving either run-time checks or static types could probably be done generically by loading in and manipulating existing type definitions.
but yeah sounds like that'd be a bit different from your approach.
